src/guides/v2.3/extension-dev-guide/extend-catalog-rule-conditions.md

```diff
@@ -0,0 +1,127 @@
+
+## Implementation of Existing Conditions
+
+In the `app/code/Magento/CatalogRule/etc/di.xml` configuration, the following `type` configuration is defined:
+
+```xml
+<type name="Magento\CatalogRule\Model\Rule\Condition\Combine">
+    <arguments>
+        <argument name="data" xsi:type="array">
+            <item name="form_name" xsi:type="string">catalog_rule_form</item>
+        </argument>
+    </arguments>
+</type>
+<type name="Magento\CatalogRule\Model\Rule\Condition\Product">
+    <arguments>
+        <argument name="data" xsi:type="array">
+            <item name="form_name" xsi:type="string">catalog_rule_form</item>
+        </argument>
+    </arguments>
+</type>
+```
+
+In the above configuration, the `catalog_rule_form` is a UI component form. It is defined in the `app/code/Magento/CatalogRule/view/adminhtml/ui_component/catalog_rule_form.xml` file.
+
+The `Magento\CatalogRule\Model\Rule\Condition\Combine` and the `Magento\CatalogRule\Model\Rule\Condition\Product` classes are important for the listed conditions and validations of the specified conditions.
+
+The `getNewChildSelectOptions` method from the `app/code/Magento/CatalogRule/Model/Rule/Condition/Combine.php` file is responsible for the listed conditions. It returns an array of the lists of valid conditions.
+
+Below is the definition of `getNewChildSelectOptions` method:
+
+```php
+/**
+ * @return array
+ */
+public function getNewChildSelectOptions()
+{
+    $productAttributes = $this->_productFactory->create()->loadAttributeOptions()->getAttributeOption();
+    $attributes = [];
+    foreach ($productAttributes as $code => $label) {
+        $attributes[] = [
+            'value' => 'Magento\CatalogRule\Model\Rule\Condition\Product|' . $code,
+            'label' => $label,
+        ];
+    }
+    $conditions = parent::getNewChildSelectOptions();
+    $conditions = array_merge_recursive(
+        $conditions,
+        [
+            [
+                'value' => \Magento\CatalogRule\Model\Rule\Condition\Combine::class,
+                'label' => __('Conditions Combination'),
+            ],
+            ['label' => __('Product Attribute'), 'value' => $attributes]
+        ]
+    );
+    return $conditions;
+}
+```
+
+In the above example, the `$conditions` array will contain the list of valid conditions. Each item in the array will have the `value` and the `label` key with the appropriate values.
+
+The `validate` method from the `app/code/Magento/CatalogRule/Model/Rule/Condition/Product.php` file is responsible for the validations of the conditions defined in the catalog price rules.
+
+Below is the definition of the `validate` method:
+
+```php
+/**
+ * Validate product attribute value for condition
+ *
+ * @param \Magento\Catalog\Model\Product|\Magento\Framework\Model\AbstractModel $model
+ * @return bool
+ */
+public function validate(\Magento\Framework\Model\AbstractModel $model)
+{
+    $attrCode = $this->getAttribute();
+    if ('category_ids' == $attrCode) {
```

Comment:
Sorry missed this one
```suggestion
    if ('category_ids' === $attrCode) {
```

Reply:
Good eye!

Reply:
@dobooth @BarnyShergold Updated the code. thanks for the review.